lib/src/layout/sidebar/sidebar_items.dart

```diff
@@ -47,26 +47,24 @@ class SidebarItems extends StatelessWidget {
   const SidebarItems({
     super.key,
     required this.items,
-    required this.currentIndex,
+    required this.currenIdentifier,
     required this.onChanged,
     this.itemSize = SidebarItemSize.medium,
     this.scrollController,
     this.selectedColor,
     this.unselectedColor,
     this.shape,
     this.cursor = SystemMouseCursors.basic,
-  }) : assert(currentIndex >= 0);
+  });
 
   /// The [SidebarItem]s used by the sidebar. If no items are provided,
   /// the sidebar is not rendered.
   final List<SidebarItem> items;
 
-  /// The current selected index. It must be in the range of 0 to
-  /// [items.length]
-  final int currentIndex;
+  final String? currenIdentifier;
```

Comment:
Why the change from indexes to unique string ids? I don't necessarily have a problem with it, but if we want to get these changes upstreamed there needs to be a reason for changing the API.

Reply:
Previously the use of `int` was due to the fact that the package only kept track of the index of the item, and given that it could only ever be one level deep the structure worked as follow
- Regular Item (1)
- Folder
   - Regular Item (2)
   - Regular Item (3)
- Regular Item (4)

This breaks down with multiple nested objects so a unique identifier for each is necessary. In this instance i made the use of string as they seemed for more fitting. Now we can allow for (n) nested items as follow:

- Regular Item (id)
- Folder
   - Regular Item (id)
   - Regular Item (id)
   - Folder
        - Regular Item (id)
        - Regular Item (id)
- Regular Item (id)